### PR TITLE
Organize files

### PR DESCRIPTION
**Pull request by timjrobinson**
I was getting frustrated by the random placement of files, this project kind of grew organically so I never took a step back to figure out the best way to structure it all. 

While we have a moment between updates I thought I'd reorganise it and make it similar in structure to the beets API so we can more easily work with both API's and copy/move components between them.

Main changes:
- Most components are now under src/modules named for what they do
- Types and Util functions have been moved into their relevant module instead of a generic file. 
- Imports can now use `@/modules` like the frontend instead of relative paths
- Added config files instead of config being randomly placed. 

There's no functionality changes in this PR, just re-organizing everything.

### CODE REVIEW

File: src/modules/chain-data/onchain.spec.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { PoolType, SubgraphPoolBase } from '@balancer-labs/sdk';
-import { Pool } from '../types';
+import { Pool } from '@/modules/pools';
 import { fetchTokens, fetchPoolsFromChain } from './onchain';
-import { subgraphPoolBase } from '../../test/factories/sor';
-import { poolBase } from '../../test/factories/pools';
+import { subgraphPoolBase } from '../../../tests/factories/sor';
+import { poolBase } from '../../../tests/factories/pools';
```
Comment (alter-eggo) [inline src/modules/chain-data/onchain.spec.ts:5]: `@/tests` is probably better?

File: src/modules/pools/utils.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      ...{
+        decimals: poolToken.decimals || 18,
+        priceRate: poolToken.priceRate || null,
+        weight: poolToken.weight || null,
+      },
+    };
+  });
+  return {
+    ...pool,
+    ...{ tokens },
```
Comment (alter-eggo) [inline src/modules/pools/utils.ts:25]: hm, why not just `tokens`

**Review by alter-eggo — APPROVED** [review]
some small comments

File: src/modules/pools/utils.ts (hunk 2 above)
Comment (timjrobinson) [reply src/modules/pools/utils.ts:25]: Good point. Fixed

File: src/modules/chain-data/onchain.spec.ts (hunk 1 above)
Comment (timjrobinson) [reply src/modules/chain-data/onchain.spec.ts:5]: Good idea, makes things much nicer.

**Pull request merged**